core/src/main/clojure/xtdb/pgwire.clj

```diff
@@ -1878,6 +1892,10 @@
           (try
             (let [conn-socket (.accept accept-socket)]
               (.setTcpNoDelay conn-socket true)
+              (.setSendBufferSize conn-socket socket-buffer-size)
```

Comment:
potentially silly question on my part - do we need both `.setSendBufferSize` and `BufferedOutputStream` here? i.e. are they both having the same effect?

Reply:
Yes, maybe I should do a run with just these and also check the underlying implementation.

Reply:
From the docs
```java
    /**
     * Sets the {@link SocketOptions#SO_SNDBUF SO_SNDBUF} option to the
     * specified value for this {@code Socket}.
     * The {@link SocketOptions#SO_SNDBUF SO_SNDBUF} option is used by the
     * platform's networking code as a hint for the size to set the underlying
     * network I/O buffers.
```
It seems to get honored on Linux. With just `setBufferSize` and without `BufferedOutputStream` I don't see any performance reduction. So maybe we just set that.

Reply:
It seems that 
```
socket.setSendBufferSize(bufferSize);
socket.setReceiveBufferSize(bufferSize);
``` 
has no effect for performance on the ingest benchmark. I guess this is more on tcp socket level.